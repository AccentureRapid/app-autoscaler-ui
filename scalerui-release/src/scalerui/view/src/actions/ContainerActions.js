import ActionTypes from '../constants/ActionTypes'

let ContainerActions  = {

  setAppId(appId) {
    return {
      type: ActionTypes.SET_CONTAINER_APP_ID,
      appId
    }
  },

  setServiceId(serviceId) {
    return {
      type: ActionTypes.SET_CONTAINER_SERVICE_ID,
      serviceId
    }
  },

  setShowHeaderBar(show) {
    return {
      type: ActionTypes.SET_CONTAINER_SHOW_HEADER_BAR,
      show
    }
  },

}

export default ContainerActions
