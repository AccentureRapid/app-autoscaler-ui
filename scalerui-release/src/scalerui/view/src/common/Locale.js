import zh from 'react-intl/locale-data/zh'
import en from 'react-intl/locale-data/en'
import zh_CN from '../constants/AutoscalingNLS/nls/zh/ScalingMessages'
import en_US from '../constants/AutoscalingNLS/nls/ScalingMessages'
import { addLocaleData } from 'react-intl'

addLocaleData([...en, ...zh])

let Locale = {
  getLocale(language) {
    switch (language) {
      case 'en-US':
        return en_US
      case 'zh-CN':
        return zh_CN
      default:
        return en_US
    }
  }
}

export default Locale
