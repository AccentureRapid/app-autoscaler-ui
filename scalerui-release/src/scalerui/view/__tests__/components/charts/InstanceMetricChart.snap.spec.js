import MetricChart from '../../../src/components/charts/MetricChart'
import renderer from 'react-test-renderer'
import store from '../../../src/store'
import fs from 'fs'
import React from 'react'
import TestConstants from '../../TestConstants'
import { IntlProvider, intlShape } from 'react-intl'
import Locale from '../../../src/common/Locale'
import MetricsActions from '../../../src/actions/MetricsActions'
import ContainerActions from '../../../src/actions/ContainerActions'
import moment from 'moment'
import Util from '../../../src/common/Util'
import PolicyActions from '../../../src/actions/PolicyActions'

const policyFile = '__tests__/assets/policy/policy.json'
const metricsFile = '__tests__/assets/metrics/memoryused.json'
let policyData = JSON.parse(fs.readFileSync(policyFile))
let metricsData = JSON.parse(fs.readFileSync(metricsFile))

const startTime = moment(parseInt(metricsData.resources[0].collected_at / 1000000))
const endTime = moment(parseInt(metricsData.resources[metricsData.resources.length - 1].collected_at / 1000000))
const props = {
  selectedIndex: 0,
  metricName: 'memoryused',
  fromTime: startTime,
  toTime: endTime,
  refresh: false,
  intl: intlShape.isRequired,
}
const query = {
  'start-time': startTime * 1000000,
  'end-time': endTime * 1000000,
  'page': 1,
  'results-per-page': 10000000,
  'order': 'asc'
}

metricsData = JSON.parse(fs.readFileSync(metricsFile)).resources
const basicInfo = MetricsActions.getBasicInfo('memoryused', metricsData)
const interval = basicInfo.interval
const maxIndex = basicInfo.maxIndex
const unit = basicInfo.unit
const maxValue = basicInfo.maxValue
const transformedData = MetricsActions.transform(metricsData, interval, metricsData[0].collected_at, metricsData[metricsData.length - 1].collected_at)
store.dispatch(MetricsActions.getInstanceMetricsSuccess('memoryused', unit, maxIndex, maxValue, transformedData, query))
store.dispatch(ContainerActions.setAppId(TestConstants.appId))

describe('MetricChart', () => {
  const wrapper = renderer.create(
    <IntlProvider locale={Util.getDefaultLocal()} messages={Locale.getLocale(Util.getDefaultLocal())}>
      <MetricChart store={store} {...props} />
    </IntlProvider>
  )

  it.only('renders MetricChart component correctly -- no upper no lower', () => {
    const tree = wrapper.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it.only('renders MetricChart component correctly -- no upper', () => {
    let mapPolicy = Util.transformArrayToMap(policyData)
    delete mapPolicy.scaling_rules_map.memoryused['upper']
    store.dispatch(PolicyActions.getPolicySuccess(mapPolicy))
    const tree = wrapper.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it.only('renders MetricChart component correctly -- no lower', () => {
    let mapPolicy = Util.transformArrayToMap(policyData)
    delete mapPolicy.scaling_rules_map.memoryused['lower']
    store.dispatch(PolicyActions.getPolicySuccess(mapPolicy))
    const tree = wrapper.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it.only('renders MetricChart component correctly', () => {
    let mapPolicy = Util.transformArrayToMap(policyData)
    store.dispatch(PolicyActions.getPolicySuccess(mapPolicy))
    const tree = wrapper.toJSON()
    expect(tree).toMatchSnapshot()
  })

  afterAll(() => {
    wrapper.unmount()
  })

})
