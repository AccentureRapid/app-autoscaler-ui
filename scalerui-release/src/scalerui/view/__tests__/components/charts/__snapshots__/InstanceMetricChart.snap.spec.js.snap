// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetricChart renders MetricChart component correctly -- no lower 1`] = `
<div>
  <div
    className="bx--row"
  >
    <div
      className="bx--col-xs-12 bx--col-md-4"
    >
      <label>
        {"option":{"title":{"left":"center","text":"Memory Used"},"series":[{"type":"gauge","data":[25],"startAngle":170,"endAngle":10,"radius":"80%","axisLine":{"lineStyle":{"color":[[0.3333333333333333,"#5aa700"],[0.4444444444444444,"#ffaa00"],[0.7407407407407407,"#ff5500"],[1,"#ff0000"]],"opacity":0.6},"width":50},"center":["50%","65%"],"splitLine":{"lineStyle":{"width":1}},"axisTick":{"show":false},"axisLabel":{"show":true,"distance":-53,"fontSize":11},"detail":{"formatter":"{value} MB"},"max":270}]}}
      </label>
    </div>
    <div
      className="bx--col-xs-12 bx--col-md-7"
    >
      <label>
        {"option":{"title":{"left":"center","text":"Memory Used","subtext":"2018-05-15 17:00:42 ~ 2018-05-15 17:53:57"},"tooltip":{"trigger":"axis"},"toolbox":{"show":true,"right":20,"feature":{"dataView":{"title":"Raw Data","readOnly":false,"lang":["Raw Data","Close","Update"]},"saveAsImage":{"title":"Save Image","type":"png","lang":["Click to Save"]}}},"grid":{"top":60,"left":40,"right":200,"bottom":60},"xAxis":[{"type":"category","boundaryGap":true,"data":["17:00:41","17:00:56","17:01:11","17:01:26","17:01:41","17:01:56","17:02:11","17:02:26","17:02:41","17:02:56","17:03:11","17:03:26","17:03:41","17:03:56","17:04:11","17:04:26","17:04:41","17:04:56","17:05:11","17:05:26","17:05:41","17:05:56","17:06:11","17:06:26","17:06:41","17:06:56","17:07:11","17:07:26","17:07:41","17:07:56","17:08:11","17:08:26","17:08:41","17:08:56","17:09:11","17:09:26","17:09:41","17:09:56","17:10:11","17:10:26","17:10:41","17:10:56","17:11:11","17:11:26","17:11:41","17:11:56","17:12:11","17:12:26","17:12:41","17:12:56","17:13:11","17:13:26","17:13:41","17:13:56","17:14:11","17:14:26","17:14:41","17:14:56","17:15:11","17:15:26","17:15:41","17:15:56","17:16:11","17:16:26","17:16:41","17:16:56","17:17:11","17:17:26","17:17:41","17:17:56","17:18:11","17:18:26","17:18:41","17:18:56","17:19:11","17:19:26","17:19:41","17:19:56","17:20:11","17:20:26","17:20:41","17:20:56","17:21:11","17:21:26","17:21:41","17:21:56","17:22:11","17:22:26","17:22:41","17:22:56","17:23:11","17:23:26","17:23:41","17:23:56","17:24:11","17:24:26","17:24:41","17:24:56","17:25:11","17:25:26","17:25:41","17:25:56","17:26:11","17:26:26","17:26:41","17:26:56","17:27:11","17:27:26","17:27:41","17:27:56","17:28:11","17:28:26","17:28:41","17:28:56","17:29:11","17:29:26","17:29:41","17:29:56","17:30:11","17:30:26","17:30:41","17:30:56","17:31:11","17:31:26","17:31:41","17:31:56","17:32:11","17:32:26","17:32:41","17:32:56","17:33:11","17:33:26","17:33:41","17:33:56","17:34:11","17:34:26","17:34:41","17:34:56","17:35:11","17:35:26","17:35:41","17:35:56","17:36:11","17:36:26","17:36:41","17:36:56","17:37:11","17:37:26","17:37:41","17:37:56","17:38:11","17:38:26","17:38:41","17:38:56","17:39:11","17:39:26","17:39:41","17:39:56","17:40:11","17:40:26","17:40:41","17:40:56","17:41:11","17:41:26","17:41:41","17:41:56","17:42:11","17:42:26","17:42:41","17:42:56","17:43:11","17:43:26","17:43:41","17:43:56","17:44:11","17:44:26","17:44:41","17:44:56","17:45:11","17:45:26","17:45:41","17:45:56","17:46:11","17:46:26","17:46:41","17:46:56","17:47:11","17:47:26","17:47:41","17:47:56","17:48:11","17:48:26","17:48:41","17:48:56","17:49:11","17:49:26","17:49:41","17:49:56","17:50:11","17:50:26","17:50:41","17:50:56","17:51:11","17:51:26","17:51:41","17:51:56","17:52:11","17:52:26","17:52:41","17:52:56","17:53:11","17:53:26","17:53:41","17:53:56"],"axisLabel":{"rotate":45}}],"yAxis":[{"type":"value","scale":true,"min":0,"splitLine":{"show":true,"lineStyle":{"opacity":0.3}},"name":"MB","max":270}],"visualMap":{"show":false,"pieces":[{"color":"#5aa700","colorAlpha":"60%","gt":0,"lte":90},{"color":"#ff0000","colorAlpha":"60%","gt":200,"lte":270},{"color":"#ff5500","colorAlpha":"60%","gt":120,"lte":200},{"color":"#ffaa00","colorAlpha":"60%","gt":90,"lte":120}]},"series":[{"name":"Memory Used","type":"bar","barGap":0,"itemStyle":{"normal":{"barBorderRadius":4}},"animationEasing":"elasticOut","data":[20,20,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,20,20,20,20,20,20,20,20,20,20,20,19,19,19,19,19,19,19,19,19,19,19,19,19,19,20,20,20,20,20,20,20,20,20,20,20,20,19,19,19,19,19,19,19,19,19,19,20,20,20,20,20,20,20,20,20,20,20,20,20,20,20,19,19,19,19,19,19,19,19,19,19,19,20,20,20,20,19,19,20,20,20,20,20,20,20,20,20,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,20,20,20,20,20,20,20,20,20,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,20,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,23,32,41,41,47,57,57,60,59,59,60,61,62,62,63,63,62,63,63,60,null,27,26,25,23,25,25,23,24,22,23,22,21,21,22,24,24,26,24,25],"areaStyle":{},"markLine":{"silent":true,"lineStyle":{"type":"dotted","width":1.5},"data":[{"yAxis":200,"lineStyle":{"color":"#ff0000"},"label":{"formatter":"+4 if Memory Used &gt;= 200"}},{"yAxis":120,"lineStyle":{"color":"#ff5500"},"label":{"formatter":"+3 if Memory Used &gt; 120"}},{"yAxis":90,"lineStyle":{"color":"#ffaa00"},"label":{"formatter":"+2 if Memory Used &gt;= 90"}}]}}]},"style":{"height":300}}
      </label>
    </div>
    <div
      className="bx--col-xs-12 bx--col-md-1"
    />
  </div>
</div>
`;

exports[`MetricChart renders MetricChart component correctly -- no upper 1`] = `
<div>
  <div
    className="bx--row"
  >
    <div
      className="bx--col-xs-12 bx--col-md-4"
    >
      <label>
        {"option":{"title":{"left":"center","text":"Memory Used"},"series":[{"type":"gauge","data":[25],"startAngle":170,"endAngle":10,"radius":"80%","axisLine":{"lineStyle":{"color":[[0.14285714285714285,"#3388ff"],[0.42857142857142855,"#33ccff"],[1,"#5aa700"]],"opacity":0.6},"width":50},"center":["50%","65%"],"splitLine":{"lineStyle":{"width":1}},"axisTick":{"show":false},"axisLabel":{"show":true,"distance":-53,"fontSize":11},"detail":{"formatter":"{value} MB"},"max":70}]}}
      </label>
    </div>
    <div
      className="bx--col-xs-12 bx--col-md-7"
    >
      <label>
        {"option":{"title":{"left":"center","text":"Memory Used","subtext":"2018-05-15 17:00:42 ~ 2018-05-15 17:53:57"},"tooltip":{"trigger":"axis"},"toolbox":{"show":true,"right":20,"feature":{"dataView":{"title":"Raw Data","readOnly":false,"lang":["Raw Data","Close","Update"]},"saveAsImage":{"title":"Save Image","type":"png","lang":["Click to Save"]}}},"grid":{"top":60,"left":40,"right":200,"bottom":60},"xAxis":[{"type":"category","boundaryGap":true,"data":["17:00:41","17:00:56","17:01:11","17:01:26","17:01:41","17:01:56","17:02:11","17:02:26","17:02:41","17:02:56","17:03:11","17:03:26","17:03:41","17:03:56","17:04:11","17:04:26","17:04:41","17:04:56","17:05:11","17:05:26","17:05:41","17:05:56","17:06:11","17:06:26","17:06:41","17:06:56","17:07:11","17:07:26","17:07:41","17:07:56","17:08:11","17:08:26","17:08:41","17:08:56","17:09:11","17:09:26","17:09:41","17:09:56","17:10:11","17:10:26","17:10:41","17:10:56","17:11:11","17:11:26","17:11:41","17:11:56","17:12:11","17:12:26","17:12:41","17:12:56","17:13:11","17:13:26","17:13:41","17:13:56","17:14:11","17:14:26","17:14:41","17:14:56","17:15:11","17:15:26","17:15:41","17:15:56","17:16:11","17:16:26","17:16:41","17:16:56","17:17:11","17:17:26","17:17:41","17:17:56","17:18:11","17:18:26","17:18:41","17:18:56","17:19:11","17:19:26","17:19:41","17:19:56","17:20:11","17:20:26","17:20:41","17:20:56","17:21:11","17:21:26","17:21:41","17:21:56","17:22:11","17:22:26","17:22:41","17:22:56","17:23:11","17:23:26","17:23:41","17:23:56","17:24:11","17:24:26","17:24:41","17:24:56","17:25:11","17:25:26","17:25:41","17:25:56","17:26:11","17:26:26","17:26:41","17:26:56","17:27:11","17:27:26","17:27:41","17:27:56","17:28:11","17:28:26","17:28:41","17:28:56","17:29:11","17:29:26","17:29:41","17:29:56","17:30:11","17:30:26","17:30:41","17:30:56","17:31:11","17:31:26","17:31:41","17:31:56","17:32:11","17:32:26","17:32:41","17:32:56","17:33:11","17:33:26","17:33:41","17:33:56","17:34:11","17:34:26","17:34:41","17:34:56","17:35:11","17:35:26","17:35:41","17:35:56","17:36:11","17:36:26","17:36:41","17:36:56","17:37:11","17:37:26","17:37:41","17:37:56","17:38:11","17:38:26","17:38:41","17:38:56","17:39:11","17:39:26","17:39:41","17:39:56","17:40:11","17:40:26","17:40:41","17:40:56","17:41:11","17:41:26","17:41:41","17:41:56","17:42:11","17:42:26","17:42:41","17:42:56","17:43:11","17:43:26","17:43:41","17:43:56","17:44:11","17:44:26","17:44:41","17:44:56","17:45:11","17:45:26","17:45:41","17:45:56","17:46:11","17:46:26","17:46:41","17:46:56","17:47:11","17:47:26","17:47:41","17:47:56","17:48:11","17:48:26","17:48:41","17:48:56","17:49:11","17:49:26","17:49:41","17:49:56","17:50:11","17:50:26","17:50:41","17:50:56","17:51:11","17:51:26","17:51:41","17:51:56","17:52:11","17:52:26","17:52:41","17:52:56","17:53:11","17:53:26","17:53:41","17:53:56"],"axisLabel":{"rotate":45}}],"yAxis":[{"type":"value","scale":true,"min":0,"splitLine":{"show":true,"lineStyle":{"opacity":0.3}},"name":"MB","max":70}],"visualMap":{"show":false,"pieces":[{"color":"#33ccff","colorAlpha":"60%","gt":10,"lte":30},{"color":"#3388ff","colorAlpha":"60%","gt":0,"lte":10},{"color":"#5aa700","colorAlpha":"60%","gt":30,"lte":70}]},"series":[{"name":"Memory Used","type":"bar","barGap":0,"itemStyle":{"normal":{"barBorderRadius":4}},"animationEasing":"elasticOut","data":[20,20,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,20,20,20,20,20,20,20,20,20,20,20,19,19,19,19,19,19,19,19,19,19,19,19,19,19,20,20,20,20,20,20,20,20,20,20,20,20,19,19,19,19,19,19,19,19,19,19,20,20,20,20,20,20,20,20,20,20,20,20,20,20,20,19,19,19,19,19,19,19,19,19,19,19,20,20,20,20,19,19,20,20,20,20,20,20,20,20,20,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,20,20,20,20,20,20,20,20,20,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,20,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,23,32,41,41,47,57,57,60,59,59,60,61,62,62,63,63,62,63,63,60,null,27,26,25,23,25,25,23,24,22,23,22,21,21,22,24,24,26,24,25],"areaStyle":{},"markLine":{"silent":true,"lineStyle":{"type":"dotted","width":1.5},"data":[{"yAxis":30,"lineStyle":{"color":"#33ccff"},"label":{"formatter":"-1 if Memory Used &lt; 30"}},{"yAxis":10,"lineStyle":{"color":"#3388ff"},"label":{"formatter":"-2 if Memory Used &lt;= 10"}}]}}]},"style":{"height":300}}
      </label>
    </div>
    <div
      className="bx--col-xs-12 bx--col-md-1"
    />
  </div>
</div>
`;

exports[`MetricChart renders MetricChart component correctly -- no upper no lower 1`] = `
<div>
  <div
    className="bx--row"
  >
    <div
      className="bx--col-xs-12 bx--col-md-4"
    >
      <label>
        {"option":{"title":{"left":"center","text":"Memory Used"},"series":[{"type":"gauge","data":[25],"startAngle":170,"endAngle":10,"radius":"80%","axisLine":{"lineStyle":{"color":[[1,"#5aa700"]],"opacity":0.6},"width":50},"center":["50%","65%"],"splitLine":{"lineStyle":{"width":1}},"axisTick":{"show":false},"axisLabel":{"show":true,"distance":-53,"fontSize":11},"detail":{"formatter":"{value} MB"},"max":70}]}}
      </label>
    </div>
    <div
      className="bx--col-xs-12 bx--col-md-7"
    >
      <label>
        {"option":{"title":{"left":"center","text":"Memory Used","subtext":"2018-05-15 17:00:42 ~ 2018-05-15 17:53:57"},"tooltip":{"trigger":"axis"},"toolbox":{"show":true,"right":20,"feature":{"dataView":{"title":"Raw Data","readOnly":false,"lang":["Raw Data","Close","Update"]},"saveAsImage":{"title":"Save Image","type":"png","lang":["Click to Save"]}}},"grid":{"top":60,"left":40,"right":200,"bottom":60},"xAxis":[{"type":"category","boundaryGap":true,"data":["17:00:41","17:00:56","17:01:11","17:01:26","17:01:41","17:01:56","17:02:11","17:02:26","17:02:41","17:02:56","17:03:11","17:03:26","17:03:41","17:03:56","17:04:11","17:04:26","17:04:41","17:04:56","17:05:11","17:05:26","17:05:41","17:05:56","17:06:11","17:06:26","17:06:41","17:06:56","17:07:11","17:07:26","17:07:41","17:07:56","17:08:11","17:08:26","17:08:41","17:08:56","17:09:11","17:09:26","17:09:41","17:09:56","17:10:11","17:10:26","17:10:41","17:10:56","17:11:11","17:11:26","17:11:41","17:11:56","17:12:11","17:12:26","17:12:41","17:12:56","17:13:11","17:13:26","17:13:41","17:13:56","17:14:11","17:14:26","17:14:41","17:14:56","17:15:11","17:15:26","17:15:41","17:15:56","17:16:11","17:16:26","17:16:41","17:16:56","17:17:11","17:17:26","17:17:41","17:17:56","17:18:11","17:18:26","17:18:41","17:18:56","17:19:11","17:19:26","17:19:41","17:19:56","17:20:11","17:20:26","17:20:41","17:20:56","17:21:11","17:21:26","17:21:41","17:21:56","17:22:11","17:22:26","17:22:41","17:22:56","17:23:11","17:23:26","17:23:41","17:23:56","17:24:11","17:24:26","17:24:41","17:24:56","17:25:11","17:25:26","17:25:41","17:25:56","17:26:11","17:26:26","17:26:41","17:26:56","17:27:11","17:27:26","17:27:41","17:27:56","17:28:11","17:28:26","17:28:41","17:28:56","17:29:11","17:29:26","17:29:41","17:29:56","17:30:11","17:30:26","17:30:41","17:30:56","17:31:11","17:31:26","17:31:41","17:31:56","17:32:11","17:32:26","17:32:41","17:32:56","17:33:11","17:33:26","17:33:41","17:33:56","17:34:11","17:34:26","17:34:41","17:34:56","17:35:11","17:35:26","17:35:41","17:35:56","17:36:11","17:36:26","17:36:41","17:36:56","17:37:11","17:37:26","17:37:41","17:37:56","17:38:11","17:38:26","17:38:41","17:38:56","17:39:11","17:39:26","17:39:41","17:39:56","17:40:11","17:40:26","17:40:41","17:40:56","17:41:11","17:41:26","17:41:41","17:41:56","17:42:11","17:42:26","17:42:41","17:42:56","17:43:11","17:43:26","17:43:41","17:43:56","17:44:11","17:44:26","17:44:41","17:44:56","17:45:11","17:45:26","17:45:41","17:45:56","17:46:11","17:46:26","17:46:41","17:46:56","17:47:11","17:47:26","17:47:41","17:47:56","17:48:11","17:48:26","17:48:41","17:48:56","17:49:11","17:49:26","17:49:41","17:49:56","17:50:11","17:50:26","17:50:41","17:50:56","17:51:11","17:51:26","17:51:41","17:51:56","17:52:11","17:52:26","17:52:41","17:52:56","17:53:11","17:53:26","17:53:41","17:53:56"],"axisLabel":{"rotate":45}}],"yAxis":[{"type":"value","scale":true,"min":0,"splitLine":{"show":true,"lineStyle":{"opacity":0.3}},"name":"MB","max":70}],"visualMap":{"show":false,"pieces":[{"color":"#5aa700","colorAlpha":"60%","gt":0,"lte":70}]},"series":[{"name":"Memory Used","type":"bar","barGap":0,"itemStyle":{"normal":{"barBorderRadius":4}},"animationEasing":"elasticOut","data":[20,20,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,20,20,20,20,20,20,20,20,20,20,20,19,19,19,19,19,19,19,19,19,19,19,19,19,19,20,20,20,20,20,20,20,20,20,20,20,20,19,19,19,19,19,19,19,19,19,19,20,20,20,20,20,20,20,20,20,20,20,20,20,20,20,19,19,19,19,19,19,19,19,19,19,19,20,20,20,20,19,19,20,20,20,20,20,20,20,20,20,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,20,20,20,20,20,20,20,20,20,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,20,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,23,32,41,41,47,57,57,60,59,59,60,61,62,62,63,63,62,63,63,60,null,27,26,25,23,25,25,23,24,22,23,22,21,21,22,24,24,26,24,25],"areaStyle":{},"markLine":{"silent":true,"lineStyle":{"type":"dotted","width":1.5},"data":[]}}]},"style":{"height":300}}
      </label>
    </div>
    <div
      className="bx--col-xs-12 bx--col-md-1"
    />
  </div>
</div>
`;

exports[`MetricChart renders MetricChart component correctly 1`] = `
<div>
  <div
    className="bx--row"
  >
    <div
      className="bx--col-xs-12 bx--col-md-4"
    >
      <label>
        {"option":{"title":{"left":"center","text":"Memory Used"},"series":[{"type":"gauge","data":[25],"startAngle":170,"endAngle":10,"radius":"80%","axisLine":{"lineStyle":{"color":[[0.041666666666666664,"#3388ff"],[0.125,"#33ccff"],[0.375,"#5aa700"],[0.5,"#ffaa00"],[0.8333333333333334,"#ff5500"],[1,"#ff0000"]],"opacity":0.6},"width":50},"center":["50%","65%"],"splitLine":{"lineStyle":{"width":1}},"axisTick":{"show":false},"axisLabel":{"show":true,"distance":-53,"fontSize":11},"detail":{"formatter":"{value} MB"},"max":240}]}}
      </label>
    </div>
    <div
      className="bx--col-xs-12 bx--col-md-7"
    >
      <label>
        {"option":{"title":{"left":"center","text":"Memory Used","subtext":"2018-05-15 17:00:42 ~ 2018-05-15 17:53:57"},"tooltip":{"trigger":"axis"},"toolbox":{"show":true,"right":20,"feature":{"dataView":{"title":"Raw Data","readOnly":false,"lang":["Raw Data","Close","Update"]},"saveAsImage":{"title":"Save Image","type":"png","lang":["Click to Save"]}}},"grid":{"top":60,"left":40,"right":200,"bottom":60},"xAxis":[{"type":"category","boundaryGap":true,"data":["17:00:41","17:00:56","17:01:11","17:01:26","17:01:41","17:01:56","17:02:11","17:02:26","17:02:41","17:02:56","17:03:11","17:03:26","17:03:41","17:03:56","17:04:11","17:04:26","17:04:41","17:04:56","17:05:11","17:05:26","17:05:41","17:05:56","17:06:11","17:06:26","17:06:41","17:06:56","17:07:11","17:07:26","17:07:41","17:07:56","17:08:11","17:08:26","17:08:41","17:08:56","17:09:11","17:09:26","17:09:41","17:09:56","17:10:11","17:10:26","17:10:41","17:10:56","17:11:11","17:11:26","17:11:41","17:11:56","17:12:11","17:12:26","17:12:41","17:12:56","17:13:11","17:13:26","17:13:41","17:13:56","17:14:11","17:14:26","17:14:41","17:14:56","17:15:11","17:15:26","17:15:41","17:15:56","17:16:11","17:16:26","17:16:41","17:16:56","17:17:11","17:17:26","17:17:41","17:17:56","17:18:11","17:18:26","17:18:41","17:18:56","17:19:11","17:19:26","17:19:41","17:19:56","17:20:11","17:20:26","17:20:41","17:20:56","17:21:11","17:21:26","17:21:41","17:21:56","17:22:11","17:22:26","17:22:41","17:22:56","17:23:11","17:23:26","17:23:41","17:23:56","17:24:11","17:24:26","17:24:41","17:24:56","17:25:11","17:25:26","17:25:41","17:25:56","17:26:11","17:26:26","17:26:41","17:26:56","17:27:11","17:27:26","17:27:41","17:27:56","17:28:11","17:28:26","17:28:41","17:28:56","17:29:11","17:29:26","17:29:41","17:29:56","17:30:11","17:30:26","17:30:41","17:30:56","17:31:11","17:31:26","17:31:41","17:31:56","17:32:11","17:32:26","17:32:41","17:32:56","17:33:11","17:33:26","17:33:41","17:33:56","17:34:11","17:34:26","17:34:41","17:34:56","17:35:11","17:35:26","17:35:41","17:35:56","17:36:11","17:36:26","17:36:41","17:36:56","17:37:11","17:37:26","17:37:41","17:37:56","17:38:11","17:38:26","17:38:41","17:38:56","17:39:11","17:39:26","17:39:41","17:39:56","17:40:11","17:40:26","17:40:41","17:40:56","17:41:11","17:41:26","17:41:41","17:41:56","17:42:11","17:42:26","17:42:41","17:42:56","17:43:11","17:43:26","17:43:41","17:43:56","17:44:11","17:44:26","17:44:41","17:44:56","17:45:11","17:45:26","17:45:41","17:45:56","17:46:11","17:46:26","17:46:41","17:46:56","17:47:11","17:47:26","17:47:41","17:47:56","17:48:11","17:48:26","17:48:41","17:48:56","17:49:11","17:49:26","17:49:41","17:49:56","17:50:11","17:50:26","17:50:41","17:50:56","17:51:11","17:51:26","17:51:41","17:51:56","17:52:11","17:52:26","17:52:41","17:52:56","17:53:11","17:53:26","17:53:41","17:53:56"],"axisLabel":{"rotate":45}}],"yAxis":[{"type":"value","scale":true,"min":0,"splitLine":{"show":true,"lineStyle":{"opacity":0.3}},"name":"MB","max":240}],"visualMap":{"show":false,"pieces":[{"color":"#33ccff","colorAlpha":"60%","gt":10,"lte":30},{"color":"#3388ff","colorAlpha":"60%","gt":0,"lte":10},{"color":"#5aa700","colorAlpha":"60%","gt":30,"lte":90},{"color":"#ff0000","colorAlpha":"60%","gt":200,"lte":240},{"color":"#ff5500","colorAlpha":"60%","gt":120,"lte":200},{"color":"#ffaa00","colorAlpha":"60%","gt":90,"lte":120}]},"series":[{"name":"Memory Used","type":"bar","barGap":0,"itemStyle":{"normal":{"barBorderRadius":4}},"animationEasing":"elasticOut","data":[20,20,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,20,20,20,20,20,20,20,20,20,20,20,19,19,19,19,19,19,19,19,19,19,19,19,19,19,20,20,20,20,20,20,20,20,20,20,20,20,19,19,19,19,19,19,19,19,19,19,20,20,20,20,20,20,20,20,20,20,20,20,20,20,20,19,19,19,19,19,19,19,19,19,19,19,20,20,20,20,19,19,20,20,20,20,20,20,20,20,20,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,20,20,20,20,20,20,20,20,20,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,20,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,19,23,32,41,41,47,57,57,60,59,59,60,61,62,62,63,63,62,63,63,60,null,27,26,25,23,25,25,23,24,22,23,22,21,21,22,24,24,26,24,25],"areaStyle":{},"markLine":{"silent":true,"lineStyle":{"type":"dotted","width":1.5},"data":[{"yAxis":30,"lineStyle":{"color":"#33ccff"},"label":{"formatter":"-1 if Memory Used &lt; 30"}},{"yAxis":10,"lineStyle":{"color":"#3388ff"},"label":{"formatter":"-2 if Memory Used &lt;= 10"}},{"yAxis":200,"lineStyle":{"color":"#ff0000"},"label":{"formatter":"+4 if Memory Used &gt;= 200"}},{"yAxis":120,"lineStyle":{"color":"#ff5500"},"label":{"formatter":"+3 if Memory Used &gt; 120"}},{"yAxis":90,"lineStyle":{"color":"#ffaa00"},"label":{"formatter":"+2 if Memory Used &gt;= 90"}}]}}]},"style":{"height":300}}
      </label>
    </div>
    <div
      className="bx--col-xs-12 bx--col-md-1"
    />
  </div>
</div>
`;
